### ISSUE TITLE
"Invalid range specified" exceptions thrown

### ISSUE THREAD

**Issue by mbierlee** [intellij-dlanguage/intellij-dlanguage#234]
Regularly while typing code, the following exception is thrown:
```
Invalid range specified: (92,89); 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:132)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:222)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:217)
	at com.intellij.openapi.util.TextRange.assertProperRange(TextRange.java:213)
	at com.intellij.openapi.util.TextRange.<init>(TextRange.java:43)
	at com.intellij.openapi.util.TextRange.<init>(TextRange.java:32)
	at net.masterthought.dlanguage.highlighting.annotation.external.CompileCheck.calculateTextRange(CompileCheck.java:129)
	at net.masterthought.dlanguage.highlighting.annotation.external.CompileCheck.parseProblem(CompileCheck.java:156)
	at net.masterthought.dlanguage.highlighting.annotation.external.CompileCheck.findProblems(CompileCheck.java:73)
	at net.masterthought.dlanguage.highlighting.annotation.external.CompileCheck.checkFileSyntax(CompileCheck.java:34)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.doAnnotate(DExternalAnnotator.java:51)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.doAnnotate(DExternalAnnotator.java:21)
```

This is likely due to the fact that you're editing the file while analysis is in progress, making some ranges invalid due to the file being reduced in size meanwhile.

I suppose these exceptions should be ignored and error annotation should simply be aborted. A run after that will take care of analyzing the file in its current state.

This error is easily reproduced by opening a D file and removing all code in it, and then typing some nonsense.

**Comment by pirocks** [comment]
Is this with the current develop branch or the version of the plugin available in the repository?

**Comment by mbierlee** [comment]
Both

**Comment by pirocks** [comment]
I believe this is now fixed in develop and the fix will be coming to the next release.

**Comment by mbierlee** [comment]
Instead of invalid range errors, in its place, you now get assertion errors:
```
13:1
java.lang.AssertionError: 13:1
	at com.intellij.lang.annotation.Annotation.<init>(Annotation.java:95)
	at net.masterthought.dlanguage.highlighting.annotation.DAnnotationHolder.createAnnotation(DAnnotationHolder.java:47)
	at net.masterthought.dlanguage.highlighting.annotation.DAnnotationHolder.createErrorAnnotation(DAnnotationHolder.java:41)
	at net.masterthought.dlanguage.highlighting.annotation.external.CompileCheck$Problem.createAnnotations(CompileCheck.java:195)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.createAnnotations(DExternalAnnotator.java:72)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.apply(DExternalAnnotator.java:64)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.apply(DExternalAnnotator.java:60)
	at net.masterthought.dlanguage.highlighting.annotation.external.DExternalAnnotator.apply(DExternalAnnotator.java:20)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.applyRelevant(ExternalToolPass.java:190)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$700(ExternalToolPass.java:47)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$0(ExternalToolPass.java:164)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1160)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:160)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:337)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:327)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:277)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:291)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:246)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:246)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:417)
	at com.intellij.util.Alarm$Request.access$700(Alarm.java:344)
	at com.intellij.util.Alarm$Request$1.run(Alarm.java:384)
	at com.intellij.util.Alarm$Request.run(Alarm.java:395)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:237)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Reproducible in pretty much the exact same manner.

**Comment by pirocks** [comment]
This is actually a separate issue with a different cause. No need to create a new issue though I'll look into fixing it.

**Comment by mbierlee** [comment]
The assertion error above seems to be fixed in 1.13.2, however I can't trace it to a commit

**Comment by pirocks** [comment]
I don't remember fixing this. @SingingBush?

**Comment by SingingBush** [comment]
didn't it get fixed when I merged #236?

**Comment by pirocks** [comment]
>didn't it get fixed when I merged #236?
That fixed what this issue was originally about, however @mbierlee noticed similar but different assertion errors, which is why this issue is still open. These assertion errors then seemed to disappear in 1.13.2.

**Comment by mbierlee** [comment]
I will close this issue since the original problem is fixed. The additional problem seems to also be gone now. If it ever pops up again we can reopen/create a new issue.
